File: front/src/types/chaster.ts
export interface LockForPublic {
  _id: string;
  status: string;
  lockType: string;
  title: string;
  totalDuration: number;
  startDate: string;
  endDate: string | null;
  minDate: string;
  maxDate: string;
  minLimitDate: string | null;
  maxLimitDate: string | null;
  displayRemainingTime: boolean;
  isAllowedToViewTime: boolean;
  isReadyToUnlock: boolean;
  isFrozen: boolean;
  trusted: boolean;
  role: string;
  limitLockTime: boolean;
  combination: string;
  hideTimeLogs: boolean;
  user: any;
  keyholder: any | null;
  sharedLock: any | null;
  extensions: any[];
}

export interface PartnerSession {
  _id: string;
  slug: string;
  displayName: string;
  summary: string;
  subtitle: string;
  icon: string;
  config: any;
  mode: string;
  regularity: number;
  userData: any | null;
  nbActionsRemaining: number;
  lock: LockForPublic;
}

export interface PartnerGetSessionAuthRepDto {
  role: string;
  userId: string;
  session: PartnerSession;
}

export interface ChasterExtensionConfigSchema {
  lock_on_freeze: boolean;
  unlock_on_unfreeze: boolean;
}

export interface ChasterExtensionConfigurationSchema {
  id: string;
  role: string;
  has_linked_plugin: boolean;
  is_online: boolean;
  link_token: string | null;
  has_session: boolean;
  config: ChasterExtensionConfigSchema;
}

File: server/src/typings/chaster.py
from typing import Any, Optional
from pydantic import BaseModel, Field

class LockForPublic(BaseModel):
    id: str = Field(..., alias="_id")
    status: str = ""
    lockType: str = ""
    title: str = ""
    totalDuration: int = 0
    startDate: str = ""
    endDate: Optional[str] = None
    minDate: str = ""
    maxDate: str = ""
    minLimitDate: Optional[str] = None
    maxLimitDate: Optional[str] = None
    displayRemainingTime: bool = False
    isAllowedToViewTime: bool = False
    isReadyToUnlock: bool = False
    isFrozen: bool = False
    trusted: bool = False
    role: str = ""
    limitLockTime: bool = False
    combination: str = ""
    hideTimeLogs: bool = False
    user: Optional[Any] = None
    keyholder: Optional[Any] = None
    sharedLock: Optional[Any] = None
    extensions: list[Any] = []

class PartnerSession(BaseModel):
    id: str = Field(..., alias="_id")
    slug: str = ""
    displayName: str = ""
    summary: str = ""
    subtitle: str = ""
    icon: str = ""
    config: dict = {}
    mode: str = ""
    regularity: int = 0
    userData: Optional[Any] = None
    nbActionsRemaining: int = 0
    lock: LockForPublic

class PartnerGetSessionAuthRepDto(BaseModel):
    role: str
    userId: str
    session: PartnerSession

class PartnerConfigurationForPublic(BaseModel):
    config: dict = {}
    role: str
    user: str
    sessionId: Optional[str] = None
    extensionSlug: str
    createdAt: str
